### PR TITLE
Add support for not covered mutants

### PR DESCRIPTION
**Pull request by AlexDenisov**
By default Mull generates mutations for every function possible. There is an option (`-coverage-info`) to use the coverage information obtained from Clang (see the [docs](https://mull.readthedocs.io/en/latest/ControlMutationsTutorial.html#code-coverage) for more details) to not mutate unreachable code.

Without coverage information, Mull wastes CPU time by checking mutants that cannot be killed. With the coverage information Mull gives misleading reports saying that the mutation coverage is 100% even if only 10% of code was mutated.

This patch adds an option (`-include-not-covered`) to also include not covered mutants in the report even when code coverage is provided. The not-covered mutants will show up in the reports, but Mull won't actually mutate anything and won't run tests against not covered mutants.

This patch partially addresses #636.

P.S. This is up for review, but I'll do more testing soon as I may have missed some edge cases.

### CODE REVIEW

**Review by stanislaw — APPROVED** [review]

File: docs/generated/CLIOptions.rst [hunk 1]
```diff
@@ -44,6 +44,8 @@
 
 --coverage-info string		Path to the coverage info file (LLVM's profdata)
 
+--include-not-covered		Include (but do not run) not covered mutants. Disabled by default
```
Comment (Warwolt) [inline docs/generated/CLIOptions.rst:47]: nit: should this not say "non-covered" mutants? "not covered" sounds a little strange in plural.

Comment (AlexDenisov) [reply docs/generated/CLIOptions.rst:47]: Yeah, it makes sense. There are some contexts (like "Include (but do not run) not covered mutants." above) where "not" is more appropriate, imo. So I think it's better to stick to one form to avoid any confusion.
